sbt

```diff
@@ -277,10 +273,35 @@ acquire_sbt_jar () {
     [[ -r "$sbt_jar" ]]
   } || {
     sbt_jar="$(jar_file "$sbt_version")"
-    download_url "$(make_url "$sbt_version")" "$sbt_jar"
+    jar_url="$(make_url "$sbt_version")"
+
+    echoerr "Downloading sbt launcher for ${sbt_version}:"
+    echoerr "  From  ${jar_url}"
+    echoerr "    To  ${sbt_jar}"
+
+    download_url "${jar_url}" "${sbt_jar}"
+
+    case "${sbt_version}" in
+      0.*) vlog "SBT versions < 1.0 do not have published MD5 checksums, skipping check"; echo "" ;;
+        *) verify_sbt_jar "${sbt_jar}" ;;
+    esac
   }
 }
 
+verify_sbt_jar() {
+  local jar="${1}"
+  local md5="${jar}.md5"
+
+  download_url "$(make_url "${sbt_version}").md5" "${md5}" > /dev/null 2>&1
+  if echo "$(cat "${md5}")  ${jar}" | md5sum -c -; then
```

Comment:
General review notes / TODOs / "let's not forget"s

* What about md5sum's stdout/stderr output?
* What about if `md5sum` isn't available, but `md5` is?
* Or if none are available?

Reply:
`md5sum` is in `coreutils`, are those ever not available? macOS?

Reply:
So it looks like at the very least it's `md5sum` on both macOS and Linux (as opposed to `sha1sum` on macOS vs whatever it is for Linux).

But the only way to have the `md5sum` on macOS is by installing it (such as `brew install coreutils`).  We should decide and implement how sbt-extras should behave when it's not present.

Reply:
Is there a built-in equivalent on macOS?

Reply:
Oh!  It seems there's BSD's `md5` in `/sbin/md5`.  Probably should fall back to that.  And then still decide what to do if neither is present (e.g. on Cgywin).

Reply:
I'll do that. I imagine not having such a command means failure... Same as if the hash didn't match. After all, we could not validate the hash...